spark/src/main/java/com/netflix/iceberg/spark/source/Reader.java

```diff
@@ -406,6 +442,22 @@ private static UnsafeProjection projection(Schema finalSchema, Schema readSchema
       return iter.iterator();
     }
 
+    private InputFile openFileScan(FileScanTask task) {
+      while (!pathsToInputFiles.containsKey(task.file().path().toString())
```

Comment:
I'd probably pre-compute this map instead of building it incrementally, since it accessing next should not actually open an encryption stream. The encryption implementation of InputFile should do that when the file's stream is opened. Not a big deal though.